packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -152,38 +138,9 @@ function collectExternals(
     else if (!esmEntry.includes('node_modules')) {
       const pkgPath = resolveFrom(`${id}/package.json`, root)
       depsToTrace.add(path.dirname(pkgPath))
-    }
-    // has separate esm/require entry, assume require entry is cjs
-    else if (esmEntry !== requireEntry) {
+    } else {
```

Comment:
We could simplify these three branches, the comment above is also better removed, no?
```
    if (esmEntry && !esmEntry.includes('node_modules')) {
      // trace the dependencies of linked packages
      const pkgPath = resolveFrom(`${id}/package.json`, root)
      depsToTrace.add(path.dirname(pkgPath))
    } else {
      ssrExternals.add(id)
    }
```

Reply:
done